### PR TITLE
Initial Version of AdHocQuery

### PR DESCRIPTION
**Pull request by iamrupok**
Please Review the code for initial version of the API AdHocQuery

### CODE REVIEW

File: fineract-db/adhoq.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+CREATE TABLE `m_adhoc` (
```
Comment (nazeer1100126) [inline fineract-db/adhoq.sql:1]: This migration script should be added under fineract-provider\src\main\resources\sql\migrations\core_db. and check the existing migration scripts for filename format

File: fineract-provider/src/main/java/org/apache/fineract/commands/service/CommandWrapperBuilder.java [hunk 2]
```diff
@@ -2868,4 +2868,43 @@ public CommandWrapperBuilder deleteSmsCampaign(final Long resourceId) {
...
+        this.actionName = "CREATE";
+        this.entityName = "ADHOC";
+        this.href = "/adhoc/template";
+        return this;
+    }
+    public CommandWrapperBuilder updateAdHoc(final Long adHocId) {
+        this.actionName = "UPDATE";
+        this.entityName = "ADHOC";
+        this.entityId = adHocId;
+        this.href = "/roles/" + adHocId;
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/commands/service/CommandWrapperBuilder.java:2898]: why roles here? it should be adhoc?

File: fineract-provider/src/main/java/org/apache/fineract/adhocquery/api/AdHocApiResource.java [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+@Path("/AdhocQUery")
+@Component
+@Scope("singleton")
+public class AdHocApiResource {
+
+    /**
+     * The set of parameters that are supported in response for {@link AdhocData}
+     */
+    private final Set<String> RESPONSE_DATA_PARAMETERS = new HashSet<>(Arrays.asList("id", "name", "query", "table_name","table_field"));
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/adhocquery/api/AdHocApiResource.java:62]: RESPONSE_DATA_PARAMETERS should be id, name, query, tableName, tableFields, isActive

**Review by nazeer1100126 — CHANGES_REQUESTED** [review]

**Comment by iamrupok** [comment]
Thank you for you feedback i will update accordingly.by the way
i have added my beans appContext.xml files no error during tomcat start up
but when i try to call the api with rest client it says
not found.but other serivce are working .Any suggestion will be great help
for me

**Comment by nazeer1100126** [comment]
You need to modify appContext.xml. You need to add this new package to bean scan and you need to add repository package also. Please include this xml as part of next PR

**Comment by nazeer1100126** [comment]
@iamrupok close this PR

**Pull request closed without merging**